napari/layers/utils/_slice_input.py

```diff
@@ -47,48 +150,51 @@ def not_displayed(self) -> List[int]:
     def with_ndim(self, ndim: int) -> _SliceInput:
         """Returns a new instance with the given number of layer dimensions."""
         old_ndim = self.ndim
+        world_slice = self.world_slice.copy_with(ndim=ndim)
         if old_ndim > ndim:
-            point = self.point[-ndim:]
             order = reorder_after_dim_reduction(self.order[-ndim:])
         elif old_ndim < ndim:
-            point = (0,) * (ndim - old_ndim) + self.point
             order = tuple(range(ndim - old_ndim)) + tuple(
                 o + ndim - old_ndim for o in self.order
             )
         else:
-            point = self.point
             order = self.order
-        return _SliceInput(ndisplay=self.ndisplay, point=point, order=order)
 
-    def data_indices(
-        self, world_to_data: Affine, round_index: bool = True
-    ) -> Tuple[Union[int, slice], ...]:
-        """Transforms this into indices that can be used to slice layer data.
+        return _SliceInput(
+            ndisplay=self.ndisplay, world_slice=world_slice, order=order
+        )
+
+    def data_slice(
+        self,
+        world_to_data: Affine,
+    ) -> _ThickNDSlice[Union[float, int]]:
+        """Transforms this thick_slice into data coordinates with only relevant dimensions.
 
         The elements in non-displayed dimensions will be real numbers.
         The elements in displayed dimensions will be ``slice(None)``.
         """
         if not self.is_orthogonal(world_to_data):
             warnings.warn(
                 trans._(
-                    'Non-orthogonal slicing is being requested, but is not fully supported. Data is displayed without applying an out-of-slice rotation or shear component.',
+                    'Non-orthogonal slicing is being requested, but is not fully supported. '
+                    'Data is displayed without applying an out-of-slice rotation or shear component.',
                     deferred=True,
                 ),
                 category=UserWarning,
             )
 
         slice_world_to_data = world_to_data.set_slice(self.not_displayed)
-        world_pts = [self.point[ax] for ax in self.not_displayed]
-        data_pts = slice_world_to_data(world_pts)
-        if round_index:
-            # A round is taken to convert these values to slicing integers
```

Comment:
Is this a change in behavior for non-thick slicing of image layers? Or put another way, what happens when I have two image layers with different extents (either through shape or scale) then move the slider?

Or is this covered by the new logic in `_project_thick_slice`?

Sorry if I asked this before, but I think this is one of the potentially larger changes in this PR.

Reply:
Not sure I get the question; you mean the fact that `np.round` is no longer here?
This is now happening in `_project_thick_slice`, because we need to keep full float precision until the last moment. In that method, after adding `margins` to the `point` to get the actual begin and end of the thick slice, we round the result before using it to index the layer data.

Not sure how this relates to extents, so maybe I misunderstood the question ^^'

Reply:
I think the rounding only becomes interesting when you have two layers with different scales/shapes (which effectively determines their extents) - then the slider positions may not exactly hit the data indices of either layer and you need the rounding for things like images.

But thanks for confirming that this is happening in `_project_thick_slice`!